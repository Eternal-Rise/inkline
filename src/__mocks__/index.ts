export * from '@inkline/inkline/__mocks__/createInkline';
export * from '@inkline/inkline/__mocks__/globalProperties';
export * from '@inkline/inkline/__mocks__/localStorageMock';
export * from '@inkline/inkline/__mocks__/matchMediaMock';
export * from '@inkline/inkline/__mocks__/placeholders';
