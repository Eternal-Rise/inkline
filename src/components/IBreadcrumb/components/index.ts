export { default as IBreadcrumbItem } from './IBreadcrumbItem/IBreadcrumbItem.vue';
