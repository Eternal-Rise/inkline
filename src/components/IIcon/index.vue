<script src="./script.ts" lang="ts" />
