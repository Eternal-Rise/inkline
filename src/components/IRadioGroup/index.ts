export { default as IRadioGroup } from './IRadioGroup.vue';
export * from './mixin';
