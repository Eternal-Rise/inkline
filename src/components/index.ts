// Composition API

export { IAlert } from './IAlert';
export { IBadge } from './IBadge';
export { IBreadcrumb } from './IBreadcrumb';
export { IBreadcrumbItem } from './IBreadcrumbItem';
export { IButton } from './IButton';
export { IButtonGroup } from './IButtonGroup';
export { ICard } from './ICard';
export { ICheckbox } from './ICheckbox';
export { ICheckboxGroup } from './ICheckboxGroup';
export { ICollapsible } from './ICollapsible';
export { ICollapsibleItem } from './ICollapsibleItem';
export { IColumn } from './IColumn';
export { IContainer } from './IContainer';
export { IDropdown } from './IDropdown';
export { IDropdownDivider } from './IDropdownDivider';
export { IDropdownItem } from './IDropdownItem';
export { IForm } from './IForm';
export { IFormGroup } from './IFormGroup';
export { IFormError } from './IFormError';
export { IFormLabel } from './IFormLabel';
export { IHamburgerMenu } from './IHamburgerMenu';
export { IHeader } from './IHeader';
export { IIcon } from './IIcon';
export { IInput } from './IInput';
export { ILayout } from './ILayout';
export { ILayoutAside } from './ILayoutAside';
export { ILayoutContent } from './ILayoutContent';
export { ILayoutFooter } from './ILayoutFooter';
export { ILayoutHeader } from './ILayoutHeader';
export { IListGroup } from './IListGroup';
export { IListGroupItem } from './IListGroupItem';
export { ILoader } from './ILoader';
export { IMark } from './IMark';
export { IMedia } from './IMedia';
export { INav } from './INav';
export { INavItem } from './INavItem';
export { INavbar } from './INavbar';
export { INavbarBrand } from './INavbarBrand';
export { INavbarCollapsible } from './INavbarCollapsible';
export { INumberInput } from './INumberInput';
export { IPopover } from './IPopover';
export { IRadio } from './IRadio';
export { IRadioGroup } from './IRadioGroup';
export { IRow } from './IRow';
export { ITextarea } from './ITextarea';
export { IToggle } from './IToggle';
export { ITooltip } from './ITooltip';

// Options API
// @TODO Migrate to Composition API

export { IModal } from './IModal';
export { IPagination } from './IPagination';
export { IProgress } from './IProgress';
export { IProgressBar } from './IProgressBar';
export { ISelect } from './ISelect';
export { ISelectOption } from './ISelectOption';
export { ISidebar } from './ISidebar';
export { ITable } from './ITable';
export { ITabs } from './ITabs';
export { ITab } from './ITab';
export { ITabTitle } from './ITabTitle';
