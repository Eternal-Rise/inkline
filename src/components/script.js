const fs = require('fs');
const glob = require('glob');

glob.sync('./**/_component.scss').forEach((file) => {
    let contents = fs.readFileSync(file, 'utf8').toString();

    const componentName = contents.match(/\.([\w-]+) \{/)[1].replace('-wrapper', '');

    contents = contents.split('\n').map((line) => {
        const indent = line.match(/^\s*/)[0];
        return line
            .replace('var(----line-height)', `var(--${componentName}--line-height, var(--line-height))`)
            .replace('var(--transition-duration)', `var(--${componentName}--transition-duration, var(--transition-duration))`)
            .replace('var(--transition-easing)', `var(--${componentName}--transition-timing-function, var(--transition-timing-function))`)
            .replace(/transition-property: (.+);/, `transition-property: var(--${componentName}--transition-property, ($1));`)
            .replace('var(----color)', `var(--${componentName}--color)`)
            .replace('var(----width)', `var(--${componentName}--width)`)
            .replace('var(----height)', `var(--${componentName}--height)`)
            .replace('var(----size)', `var(--${componentName}--size)`)
            .replace('var(----z-index)', `var(--${componentName}--z-index)`)
            .replace('var(----background)', `var(--${componentName}--background)`)
            .replace('----background:', `--${componentName}--background:`)
            .replace('----border-color:', `--${componentName}--border-color:`)
            .replace('----color:', `--${componentName}--color:`)
            .replace('var(----color--disabled)', `var(--${componentName}--disabled--color)`)
            .replace('var(----background--disabled)', `var(--${componentName}--disabled--background)`)
            .replace('var(----border-color--disabled)', `var(--${componentName}--disabled--border-color)`)
            .replace('var(----font-size)', `var(--${componentName}--font-size, var(--font-size))`)
            .replace('var(----border-top-width)', `var(--${componentName}--border-top-width, var(--border-top-width))`)
            .replace('var(----border-right-width)', `var(--${componentName}--border-right-width, var(--border-right-width))`)
            .replace('var(----border-bottom-width)', `var(--${componentName}--border-bottom-width, var(--border-bottom-width))`)
            .replace('var(----border-left-width)', `var(--${componentName}--border-left-width, var(--border-left-width))`)
            .replace('var(----border-width)', `var(
${indent}${indent}--${componentName}--border-width,
${indent}${indent}(
${indent}${indent}${indent}var(--${componentName}--border-top-width, var(--border-top-width))
${indent}${indent}${indent}var(--${componentName}--border-right-width, var(--border-right-width))
${indent}${indent}${indent}var(--${componentName}--border-bottom-width, var(--border-bottom-width))
${indent}${indent}${indent}var(--${componentName}--border-left-width, var(--border-left-width))
${indent}${indent})
${indent})`)
            .replace('var(----border-top-style)', `var(--${componentName}--border-top-style, var(--border-top-style))`)
            .replace('var(----border-right-style)', `var(--${componentName}--border-right-style, var(--border-right-style))`)
            .replace('var(----border-bottom-style)', `var(--${componentName}--border-bottom-style, var(--border-bottom-style))`)
            .replace('var(----border-left-style)', `var(--${componentName}--border-left-style, var(--border-left-style))`)
            .replace('var(----border-style)', `var(
${indent}${indent}--${componentName}--border-style,
${indent}${indent}(
${indent}${indent}${indent}var(--${componentName}--border-top-style, var(--border-top-style))
${indent}${indent}${indent}var(--${componentName}--border-right-style, var(--border-right-style))
${indent}${indent}${indent}var(--${componentName}--border-bottom-style, var(--border-bottom-style))
${indent}${indent}${indent}var(--${componentName}--border-left-style, var(--border-left-style))
${indent}${indent})
${indent})`)
            .replace('var(----border-top-color)', `var(--${componentName}--border-top-color, var(--border-top-color))`)
            .replace('var(----border-right-color)', `var(--${componentName}--border-right-color, var(--border-right-color))`)
            .replace('var(----border-bottom-color)', `var(--${componentName}--border-bottom-color, var(--border-bottom-color))`)
            .replace('var(----border-left-color)', `var(--${componentName}--border-left-color, var(--border-left-color))`)
            .replace('var(----border-color)', `var(
${indent}${indent}--${componentName}--border-color,
${indent}${indent}(
${indent}${indent}${indent}var(--${componentName}--border-top-color, var(--border-top-color))
${indent}${indent}${indent}var(--${componentName}--border-right-color, var(--border-right-color))
${indent}${indent}${indent}var(--${componentName}--border-bottom-color, var(--border-bottom-color))
${indent}${indent}${indent}var(--${componentName}--border-left-color, var(--border-left-color))
${indent}${indent})
${indent})`)
            .replace('var(----box-shadow)', `var(
${indent}${indent}--${componentName}--box-shadow,
${indent}${indent}(
${indent}${indent}${indent}var(--${componentName}--box-shadow-x-offset, var(--box-shadow-offset-x))
${indent}${indent}${indent}var(--${componentName}--box-shadow-y-offset, var(--box-shadow-offset-y))
${indent}${indent}${indent}var(--${componentName}--box-shadow-blur-radius, var(--box-shadow-blur-radius))
${indent}${indent}${indent}var(--${componentName}--box-shadow-spread-radius, var(--box-shadow-spread-radius))
${indent}${indent}${indent}var(--${componentName}--box-shadow-color, var(--box-shadow-color))
${indent}${indent})
${indent})`)
            .replace('var(----border-top-left-radius)', `var(--${componentName}--border-top-left-radius, var(--border-top-left-radius))`)
            .replace('var(----border-top-right-radius)', `var(--${componentName}--border-top-right-radius, var(--border-top-right-radius))`)
            .replace('var(----border-bottom-right-radius)', `var(--${componentName}--border-bottom-right-radius, var(--border-bottom-right-radius))`)
            .replace('var(----border-bottom-left-radius)', `var(--${componentName}--border-bottom-left-radius, var(--border-bottom-left-radius))`)
            .replace('var(----border-radius)', `var(
${indent}${indent}--${componentName}--border-radius,
${indent}${indent}(
${indent}${indent}${indent}var(--${componentName}--border-top-left-radius, var(--border-top-left-radius))
${indent}${indent}${indent}var(--${componentName}--border-top-right-radius, var(--border-top-right-radius))
${indent}${indent}${indent}var(--${componentName}--border-bottom-right-radius, var(--border-bottom-right-radius))
${indent}${indent}${indent}var(--${componentName}--border-bottom-left-radius, var(--border-bottom-left-radius))
${indent}${indent})
${indent})`)
            .replace('var(----padding-top)', `var(--${componentName}--padding-top, var(--padding-top))`)
            .replace('var(----padding-right)', `var(--${componentName}--padding-right, var(--padding-right))`)
            .replace('var(----padding-bottom)', `var(--${componentName}--padding-bottom, var(--padding-bottom))`)
            .replace('var(----padding-left)', `var(--${componentName}--padding-left, var(--padding-left))`)
            .replace('var(----padding)', `var(
${indent}${indent}--${componentName}--padding,
${indent}${indent}(
${indent}${indent}${indent}var(--${componentName}--padding-top, var(--padding-top))
${indent}${indent}${indent}var(--${componentName}--padding-right, var(--padding-right))
${indent}${indent}${indent}var(--${componentName}--padding-bottom, var(--padding-bottom))
${indent}${indent}${indent}var(--${componentName}--padding-left, var(--padding-left))
${indent}${indent})
${indent})`)
            .replace('var(----margin-top)', `var(--${componentName}--margin-top, var(--margin-top))`)
            .replace('var(----margin-right)', `var(--${componentName}--margin-right, var(--margin-right))`)
            .replace('var(----margin-bottom)', `var(--${componentName}--margin-bottom, var(--margin-bottom))`)
            .replace('var(----margin-left)', `var(--${componentName}--margin-left, var(--margin-left))`)
            .replace('var(----margin)', `var(
${indent}${indent}--${componentName}--margin,
${indent}${indent}(
${indent}${indent}${indent}var(--${componentName}--margin-top, var(--margin-top))
${indent}${indent}${indent}var(--${componentName}--margin-right, var(--margin-right))
${indent}${indent}${indent}var(--${componentName}--margin-bottom, var(--margin-bottom))
${indent}${indent}${indent}var(--${componentName}--margin-left, var(--margin-left))
${indent}${indent})
${indent})`)
            .replace('var(----header--color)', `var(--${componentName}--header--color, var(--${componentName}--color))`)
            .replace('var(----body--color)', `var(--${componentName}--body--color, var(--${componentName}--color))`)
            .replace('var(----footer--color)', `var(--${componentName}--footer--color, var(--${componentName}--color))`)
            .replace('var(----header--background)', `var(--${componentName}--header--background, var(--${componentName}--background))`)
            .replace('var(----body--background)', `var(--${componentName}--body--background, var(--${componentName}--background))`)
            .replace('var(----footer--background)', `var(--${componentName}--footer--background, var(--${componentName}--background))`)
            .replace('var(----header--margin-bottom)', `var(--${componentName}--header--margin-bottom, var(--${componentName}--margin-bottom, var(--margin-bottom)))`)
            .replace('var(----footer--margin-top)', `var(--${componentName}--footer--margin-top, var(--${componentName}--margin-top, var(--margin-top)))`);
    }).join('\n');

    fs.writeFileSync(file, contents);
});
