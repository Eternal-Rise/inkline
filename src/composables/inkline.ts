import { inject } from 'vue';
import { InklineGlobalOptions } from '../plugin';

export function useInkline() {
    return inject('inkline') as InklineGlobalOptions;
}
