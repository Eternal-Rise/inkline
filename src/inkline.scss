@import 'normalize.css';
@import 'css';
