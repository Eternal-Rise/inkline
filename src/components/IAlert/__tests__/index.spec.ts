import { mount } from '@vue/test-utils';
import { IAlert } from '@inkline/inkline/components';
import { Placeholder } from '@inkline/inkline/__mocks__';

describe('Components', () => {
    describe('IAlert', () => {
        const props = {
            color: 'primary',
            size: 'md'
        };

        const wrapper = mount(IAlert, {
            props,
            slots: {
                icon: Placeholder
            }
        });

        it('should be named correctly', () => {
            expect(IAlert.__name).toEqual('IAlert');
        });

        it('should render correctly', () => {
            expect(wrapper.exists()).toBeTruthy();
            expect(wrapper.element).toMatchSnapshot();
        });

        describe('styling', () => {
            it('should add classes based on props', async () => {
                await wrapper.setProps({
                    dismissible: true
                });
                expect(wrapper.element).toHaveClass(
                    `-${props.color}`,
                    `-${props.size}`,
                    '-dismissible',
                    '-with-icon'
                );
            });
        });

        describe('functionality', () => {
            it('should hide the alert when the dismiss button is clicked', async () => {
                await wrapper.setProps({ modelValue: true, dismissible: true });
                expect(wrapper.isVisible()).toBe(true);
                await wrapper.find('span.dismiss').trigger('click');
                expect(wrapper.emitted()['update:modelValue'][0]).toEqual([false]);
                expect(wrapper.isVisible()).toBe(false);
            });
        });
    });
});
