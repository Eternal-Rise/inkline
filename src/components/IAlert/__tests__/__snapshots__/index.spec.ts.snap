// Vitest Snapshot v1

exports[`Components > IAlert > should render correctly 1`] = `
<div
  class="alert -primary -md -with-icon"
  role="alert"
>
  <span
    aria-hidden="true"
    class="icon"
    role="img"
  >
    <!--** Slot for alert icon -->
    
    <div>
      Placeholder
    </div>
    
  </span>
  <div
    class="content"
  >
    <!--** Slot for default alert content -->
    
    
  </div>
  <!--v-if-->
</div>
`;
